File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        def permutation(set_):
            temp = []

            if not set_:
                return [[]]

            for i in set_:
                copy_ = set_.copy()
                copy_.remove(i)
                res_list = permutation(copy_)
                for j in res_list:
                    temp.append(j+[i])

            return temp
        return permutation(set(nums))
            

File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if (nums.size()==1){
            return nums[0];
        }

        for (int i = nums.size()-1;i>=0;i--){
            if (i==nums.size()-1){
                continue;
            }
            if (i==nums.size()-2){
                nums[i] = max(nums[i],nums[i+1]);
            }

            else{
                nums[i] = max(nums[i]+nums[i+2],nums[i+1]);
            }
        }
        return nums[0];
        
    }

    
};
